update to modern versions of ruby and dependencies

bcm opened the pull request:
lots of changes from our fork to use ruby 1.9.3, active support 3.2, airbrake instead of hoptoad, etc.
- switch to ruby 1.9.3
- unpin dependencies to get the most modern versions
- since active support 3.2 removes its secure random module altogether, try requiring ruby 1.9's secure random before that one
- ignore Gemfile.lock as per best practive
- replace hoptoad with airbrake
- fix rake warnings and errors
- add rspec config file

there are a couple commits you probably don't want to pull (our fork-specific version bump, maybe our .rvmrc or .rspec), but there were too many other commits for me to want to open prs for each; I reckon it's faster for you to cherry pick whatever you want.

seeing as how there haven't been any changes upstream in a year, perhaps y'all aren't using this library anymore - is there a preferred alternative we should be looking at instead of extending this one? if not, we'll be adding some more features in coming days, notably the ability to propagate the key across network requests.

bcm committed:
switch to ruby 1.9.3

bcm committed:
use securerandom if provided, active support's secure random otherwise

bcm committed:
ignore Gemfile.lock as per best practice

bcm committed:
remove version pins

bcm committed:
securerandom fix for specs

bcm committed:
replace hoptoad_notifier with airbrake

bcm committed:
no seriously, get rid of Gemfile.lock

bcm committed:
fix rake warnings and errors

* pull rake in as a development dependency
* use rspec 2 package

bcm committed:
add rspec config file

bcm committed:
bump version to 0.1.0.copious.0

use 0.1.0 because of the amount of backwards-incompatible changes

bcm committed:
add gemfury

carbonfive-builds commented:
Brian-

Thanks for the pull request!  I will take a look at it tomorrow.

..mike..

On Tue, Jan 31, 2012 at 9:50 AM, Brian Moseley <
reply@reply.github.com

> wrote:
> 
> lots of changes from our fork to use ruby 1.9.3, active support 3.2,
> airbrake instead of hoptoad, etc.
> - switch to ruby 1.9.3
> - unpin dependencies to get the most modern versions
> - since active support 3.2 removes its secure random module altogether,
>   try requiring ruby 1.9's secure random before that one
> - ignore Gemfile.lock as per best practive
> - replace hoptoad with airbrake
> - fix rake warnings and errors
> - add rspec config file
> 
> there are a couple commits you probably don't want to pull (our
> fork-specific version bump, maybe our .rvmrc or .rspec), but there were too
> many other commits for me to want to open prs for each; I reckon it's
> faster for you to cherry pick whatever you want.
> 
> seeing as how there haven't been any changes upstream in a year, perhaps
> y'all aren't using this library anymore - is there a preferred alternative
> we should be looking at instead of extending this one? if not, we'll be
> adding some more features in coming days, notably the ability to propagate
> the key across network requests.
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/utahstreetlabs/log_weasel master
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/carbonfive/log_weasel/pull/1
> 
> -- Commit Summary --
> - switch to ruby 1.9.3
> - use securerandom if provided, active support's secure random otherwise
> - ignore Gemfile.lock as per best practice
> - remove version pins
> - securerandom fix for specs
> - replace hoptoad_notifier with airbrake
> - no seriously, get rid of Gemfile.lock
> - fix rake warnings and errors
> - add rspec config file
> - bump version to 0.1.0.copious.0
> 
> -- File Changes --
> 
> M .gitignore (1)
> A .rspec (1)
> M .rvmrc (4)
> D Gemfile.lock (52)
> M README.md (8)
> M Rakefile (2)
> M lib/log_weasel.rb (8)
> R lib/log_weasel/airbrake.rb (2)
> M lib/log_weasel/transaction.rb (6)
> M lib/log_weasel/version.rb (2)
> M log_weasel.gemspec (7)
> R spec/log_weasel/airbrake_spec.rb (10)
> M spec/spec_helper.rb (8)
> 
> -- Patch Links --
> 
>  https://github.com/carbonfive/log_weasel/pull/1.patch
>  https://github.com/carbonfive/log_weasel/pull/1.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/carbonfive/log_weasel/pull/1

bcm committed:
allow middleware to get key from request header

if the X-LOGWEASEL_KEY request header is provided, its value is used as the
transaction key in preference to a statically configured one.

bcm committed:
bump version to 0.1.0.copious.1

mwynholds commented:
Merged with minor differences:
- kept .rvmrc at 1.9.2, but not pegged to a patch
- version is 0.1.0
